packages/shared_preferences/shared_preferences/README.md

```diff
@@ -3,38 +3,55 @@
 [![pub package](https://img.shields.io/pub/v/shared_preferences.svg)](https://pub.dev/packages/shared_preferences)
 
 Wraps platform-specific persistent storage for simple data
-(NSUserDefaults on iOS and macOS, SharedPreferences on Android, etc.). Data may be persisted to disk asynchronously,
+(NSUserDefaults on iOS and macOS, SharedPreferences on Android, etc.). 
+Data may be persisted to disk asynchronously,
 and there is no guarantee that writes will be persisted to disk after
 returning, so this plugin must not be used for storing critical data.
 
+Supported data types are `int`, `double`, `bool`, `String` and `List<String>`.
+
 ## Usage
 To use this plugin, add `shared_preferences` as a [dependency in your pubspec.yaml file](https://flutter.dev/docs/development/platform-integration/platform-channels).
 
-### Example
-
-``` dart
-import 'package:flutter/material.dart';
-import 'package:shared_preferences/shared_preferences.dart';
-
-void main() {
-  runApp(MaterialApp(
-    home: Scaffold(
-      body: Center(
-      child: RaisedButton(
-        onPressed: _incrementCounter,
-        child: Text('Increment Counter'),
-        ),
-      ),
-    ),
-  ));
-}
-
-_incrementCounter() async {
-  SharedPreferences prefs = await SharedPreferences.getInstance();
-  int counter = (prefs.getInt('counter') ?? 0) + 1;
-  print('Pressed $counter times.');
-  await prefs.setInt('counter', counter);
-}
+### Examples
+Here are small examples that show you how to use the API.
+
+#### Write data
+```dart 
+// Obtain shared preferences.
+final prefs = await SharedPreferences.getInstance();
+
+// Save an integer value to 'counter' key. 
+await prefs.setInt('counter', 10);
+// Save an boolean value to 'repeat' key. 
+await prefs.setBool('repeat', true);
+// Save an double value to 'decimal' key. 
+await prefs.setDouble('decimal', 1.5);
+// Save an String value to 'action' key. 
+await prefs.setString('action', 'Start');
+// Save an list of strings to 'items' key. 
+await prefs.setStringList('items', <String>['Earth', 'Moon', 'Sun']);
+```
+
+#### Read data
+```dart 
+final SharedPreferences prefs = await _prefs;
```

Comment:
I don't know what `_prefs` is supposed to be here. Why is the code for getting the shared preferences different in this example from the example above?

Reply:
oh, my bad.